bin/beepmein

```diff
@@ -0,0 +1,281 @@
+#!/bin/bash
+
+# beepmein
+
+# A little wrapper around linux 'at' command, that will beep you at specific time, see --help.
+# On Debian systems make sure 'at' is installed with --no-install-recommends.
+
+# Copyright (C) 2019/2020 brontosaurusrex
+# License: GPLv3 or later.
+
+# prepare config dir
+config="$HOME/.cache/beepmein"
+mkdir -p "$config"
+
+# var
+delay="0"
+
+# softlinks to audio samples
+alarmSample="$config/alarm"     # longer sample
+reminderSample="$config/remind" # shorter sample
+
+# Will kill all children with 'killall beepmein'
+trap 'kill $(jobs -p) >/dev/null 2>&1' EXIT
+
+# required
+command -v at >/dev/null 2>&1 || { >&2 echo "I need at: sudo apt install --no-install-recommends at" ; exit 1; }
+command -v speaker-test >/dev/null 2>&1 || { >&2 echo "I need speaker-test from alsa-utils (for reminder sound)." ; exit 1; }
+command -v notify-send >/dev/null 2>&1 || { >&2 echo "I need notify-send." ; exit 1; }
+#( command -v mpv >/dev/null 2>&1 || command -v cvlc >/dev/null 2>&1 ) || { >&2 echo "I need mpv or vlc (for alarm sounds)." ; exit 1; }
+command -v yad >/dev/null 2>&1 || { >&2 echo "I need yad." ; exit 1; }
+
+# help
+help () { 
+cat << EOF
+beepmein help
+
+    Set reminder or alarm:
+
+        beepmein now + 3 minutes # reminder ('at' time format)
+        beepmein 22:10           # reminder ('at' time format)
+        beepmein 3               # reminder (in 3 minutes, beepmein shortcut),
+        beepmein 8:00 --alarm    # alarm (must be last parameter)
+        beepmein --nr 5          # with no reason (must be first parameter)
+           
+    Set sounds: (playback depends on mpv or vlc)
+
+        beepmein --set-alarm <path/to/alarm.sound>
+        beepmein --set-remind <path/to/reminder.sound>
+
+    Test playback, blinking and notifications:
+
+        beepmein --reaction reminder_text
+        beepmein --reaction --alarm alarm_text
+        beepmein --test # test everything
+        
+    List jobs
+    
+        beepmein --list # or -l
+
+    Kill all running playback/beeping:
+
+        killall beepmein 
+
+    Notes:
+
+        at -l       # will list pending jobs (or atq)
+        at -r <num> # will remove job <num>  (or atrm)
+
+EOF
+}
+
+# superecho
+superecho () {
+    if [ -t 0 ] ; then
+        echo "$1"
+    else
+        notify-send -t 3500 "$1"
+    fi
+}
+
+# reminder sound, needs speaker-test (alsa-utils)
+beep () {
+    [ -z "$1" ] && loops="1" || loops="$1"
```

Comment:
Variables scoped to the function should be declared `local`.

Reply:
Positional, immutable "$1" should be declared local?

Reply:
edit: Ok, I see what you mean down there

    somefunc() {
       local path=$1
    }